project_tier/course_materials/models.py

```diff
@@ -0,0 +1,191 @@
+from django.db import models
+from wagtail.core.models import Page
+from wagtail.admin.edit_handlers import FieldPanel, MultiFieldPanel
+from wagtail.documents.edit_handlers import DocumentChooserPanel
+from modelcluster.fields import ParentalKey
+from modelcluster.contrib.taggit import ClusterTaggableManager
+from taggit.models import Tag, TaggedItemBase
+
+"""
+Okay, the way these tags work is horrifying. The below three models aren't
+actual `Tag` subclasses, they're through-models. Eg, they represent the
+relationship between the page and each type of tag, so don't let their names
+fool you.
+
+The Wagtail docs say to do it this way:
+http://docs.wagtail.io/en/v2.0/getting_started/tutorial.html#tagging-posts
+
+Unfortunately it doesn't cover how to have multiple tags on a single page,
+which we need since we're tagging 3 different things. By default, the tags'
+reverse-accessors clash. See: https://github.com/alex/django-taggit/issues/50
+
+Per this test, we can set the `related_name` of the managers:
+https://github.com/alex/django-taggit/blob/590918f/tests/models.py#L23-L25
+Setting to `+` removes the reverse accessor clash.
+
+However, since we use ParentalKey in the through-models, we're forced to set
+a `related_name` for each relationship. The name cannot be the same as
+the managers we add to the model, so `_relationship` has been appended to them.
+
+It's messy, but perhaps the best way to achieve what we want here, short of
+changing the actual tagging API within Wagtail or django-taggit.
+"""
+
+class DisciplineTag(TaggedItemBase):
+    content_object = ParentalKey(
+        'course_materials.CourseMaterialsPage',
+        # django-modelcluster requires this to be set
+        related_name='discipline_tags_relationship',
+        on_delete=models.CASCADE
+    )
+
+
+class CourseLevelTag(TaggedItemBase):
+    content_object = ParentalKey(
+        'course_materials.CourseMaterialsPage',
+        # django-modelcluster requires this to be set
+        related_name='course_level_tags_relationship',
+        on_delete=models.CASCADE
+    )
+
+
+class ProtocolTag(TaggedItemBase):
+    content_object = ParentalKey(
+        'course_materials.CourseMaterialsPage',
+        # django-modelcluster requires this to be set
+        related_name='protocol_tags_relationship',
+        on_delete=models.CASCADE
+    )
+
+
+class CourseMaterialsPage(Page):
+    syllabus = models.ForeignKey(
+        'wagtaildocs.Document',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+
+    course_description = models.TextField()
+
+    course_title = models.TextField()
+
+    university = models.TextField()
+
+    course_name = models.TextField()
+
+    semester = models.TextField()
+
+    instructor = models.TextField()
```

Comment:
Are these fields all required? Also, since these are showing up on the front-end I think we'd preferably want to limit the length of them. See [CharField vs TextField](https://stackoverflow.com/questions/7354588/django-charfield-vs-textfield#7354680)